src/UI/Buyer/src/app/services/order/cart.service.ts

```diff
@@ -105,17 +105,21 @@ export class CartService {
       return await this.upsertLineItem(lineItem)
     }
     if (!this.initializingOrder) {
-      this.initializingOrder = true
-      if(this.userService.isAnonymous()) {
-        await this.state.createAndSetOrder(this.order)
-      } else {
-        await this.state.reset()
-      }
-      this.initializingOrder = false
+      await this.initializeOrder()
       return await this.upsertLineItem(lineItem)
     }
   }
 
+  async initializeOrder(): Promise<void> {
+    this.initializingOrder = true
+    if(this.userService.isAnonymous()) {
+      await this.state.createAndSetOrder(this.order)
+    } else {
+      await this.state.reset()
+    }
+    this.initializingOrder = false
```

Comment:
If there's an error between when its set and this line then initializingOrder will be perpetually true. I think you might want to wrap this in a try {} finally {] and set initializingOrder to false in the finally